app/src/main/java/org/fossasia/pslab/communication/sensors/MPU6050.java

```diff
@@ -0,0 +1,136 @@
+        params.put("KalmanFilter", new ArrayList(Arrays.asList(.01, .1, 1, 10, 100, 1000, 10000, null)));       //Replaced "OFF" with null.
+        setGyroRange(2000);
+        setAccelRange(16);
+        powerUp();
+    }
+
+    public void KalmanFilter(Double opt) throws IOException, NullPointerException {
+        ArrayList<double[]> noise = new ArrayList<>();
+        double[] innerNoiseArray = new double[NUMPLOTS];
+        ArrayList<Double> vals;
+        double standardDeviation;
+        if(opt == null){        //Replaced "OFF" with null.
+            K = null;
+        }
+        for (int a = 0; a < 500; a++){
+            vals = getRaw();
+            for (int b = 0; b < NUMPLOTS; b++){
+                innerNoiseArray[b] = vals.get(b);
+                noise.set(b, innerNoiseArray);
+            }
+        }
+
+        for(int a = 0; a < NUMPLOTS; a++){
+            standardDeviation = FastMath.sqrt(StatUtils.variance(noise.get(a)));        //Apachae Commons Maths used to calculate standard deviation
+            K.set(a, new KalmanFilter(1. / opt, Math.pow(standardDeviation,2)));
+        }
+
+    }
+
+    public ArrayList<Character> getVals(int addr, int bytestoread) throws IOException {
+       return i2c.readBulk(ADDRESS, addr, bytestoread);
+    }
+
+    public void powerUp() throws IOException {
+        i2c.writeBulk(ADDRESS, new int[]{0x6B,0});
+    }
+
+    public void setGyroRange(int rs) throws IOException {
+        GR = (int)params.get("setGyroRange").get(rs);
+        i2c.writeBulk(ADDRESS, new int[]{GYRO_CONFIG, GR << 3});
+    }
+
+    public void setAccelRange(int rs) throws IOException {
+        AR = (int)params.get("setAccelRange").get(rs);
+        i2c.writeBulk(ADDRESS, new int[]{ACCEL_CONFIG, AR << 3});
+    }
+
+    public ArrayList<Double> getRaw() throws IOException, NullPointerException {
+        ArrayList<Character> vals = getVals(0x3B, 14);
+        ArrayList<Double> raw = new ArrayList<>();
+        if(vals.size() == 14) {
+            for (int a = 0; a < 3; a++)
+                raw.set(a, 1. * int16(vals.get(a * 2) << 8 | vals.get(a * 2 + 1)) / ACCEL_SCALING[AR]);
+            for (int a = 4; a < 7; a++)
+                raw.set(a, 1. * int16(vals.get(a * 2) << 8 | vals.get(a * 2 + 1)) / GYRO_SCALING[GR]);
+            raw.set(3, int16(vals.get(6) << 8 | vals.get(7)) / 340. + 36.53);
+            if (K.isEmpty())
+                return raw;
+            else {
+                for (int b = 0; b < NUMPLOTS; b++) {
+                        K.get(b).inputLatestNoisyMeasurement(raw.get(b));
+                        raw.set(b, K.get(b).getLatestEstimatedMeasurement());
+                }
+                return raw;
+            }
+        }
+        return null;
+    }
+
+    public double[] getAcceleration() throws IOException {
+        ArrayList<Character> vals = getVals(0x3B,6);
+        int ax = int16(vals.get(0) << 8 | vals.get(1));
```

Comment:
I am not sure if this does anything finally.
you are giving and "int" to int16 to convert it to short and then again putting it into "int". So number of bits are same as before.
remove short, we would work with "int" everywhere. As it just a matter of number of bits and Lower bit order is same in both short and int.

P.S: when implemented something just do a dry run to get to know what its doing.